Makefile

```diff
@@ -0,0 +1,43 @@
+.PHONY: all build check clean docs format lint test \
```

Comment:
I'm not necessarily against the idea of introducing a `Makefile`. But: this is a Rust project and has the expectation that folks will be using Rust-native tooling (and, by implication, holds as a non-goal that users should be installing and using non-Rust-native tooling). So:

I would like to hear the top-line goals that lead to the introduction of the `Makefile`. Let's keep in mind that it's possible that the need could be just as well served by a shell script, or by Cargo itself (which may avoid the need to introduce one more tool to our prescribed toolchain).

Reply:
So, my next Readme update would basically say you have the option. The `Makefile` comes in nicely to handle things like working w/ `nocapture` and running different targets, i.e. `test.headless`, which are necessary to write repeatedly. Personally, I abuse `cargo watch` when developing and can run a watch against a make target that has multiple parameters. Also, it's nice for beginners to see what people tend to use, almost like documentation. 

For me, having a `Makefile` is not implying the use of other, non-rust things (we control the targets), but it's excellent for working w/ shell vars like backtrace, chromedriver, calling out to +nightly, etc.